security/cmd/istio_ca/main.go

```diff
@@ -148,6 +151,9 @@ func init() {
 			cmd.ListenedNamespaceKey+"} environment variable. If neither is set, Citadel listens to all namespaces.")
 	flags.StringVar(&opts.istioCaStorageNamespace, "citadel-storage-namespace", "istio-system", "Namespace where "+
 		"the Citadel pod is running. Will not be used if explicit file or other storage mechanism is specified.")
+	flags.BoolVar(&opts.explicitOptInRequired, "explicit-opt-in", false, "Specifies whether Citadel requires "+
+		"explicit opt-in for creating secrets. If set, only namespaces labeled with 'istio-injection=enabled' will "+
+		"have secrets created.")
```

Comment:
Please also add "this feature is only available in key and certificates delivered through secret volume mount".

Reply:
added. Will be in next push.